homeassistant/components/lookin/media_player.py

```diff
@@ -86,6 +87,29 @@ def __init__(
         for function_name, feature in _FUNCTION_NAME_TO_FEATURE.items():
             if function_name in self._function_names:
                 self._attr_supported_features |= feature
+        self._source_list: dict[str, str] | None = None
+
+    @property
+    def source_list(self) -> list[str]:
+        """List of available input sources."""
+        return list(self._source_list.keys()) if self._source_list else []
+
+    async def async_select_source(self, source: str) -> None:
+        """Choose an available playlist and play it."""
+        if not self._source_list:
+            return
+        await self._async_send_command(command="mode", signal=self._source_list[source])
+
+    async def async_added_to_hass(self) -> None:
+        """Get list of available input sources."""
+        if self._source_list is None and "mode" in self._function_names:
+            if sources := await self._lookin_protocol.get_media_sources(
+                uuid=self._uuid
+            ):
```

Comment:
You could combine this if to save one level of indent

Reply:
With formatting though its not much better

```python
        if (
            self._source_list is None
            and "mode" in self._function_names
            and (
                sources := await self._lookin_protocol.get_media_sources(
                    uuid=self._uuid
                )
            )
        ):
```

Reply:
one level is always better, but this solution is much harder to read